recipes/boost/all/conanfile.py

```diff
@@ -555,21 +666,24 @@ def _build_flags(self):
             flags.append("boost.locale.iconv=on boost.locale.icu=off")
         else:
             flags.append("boost.locale.iconv=off boost.locale.icu=off")
-            flags.append("--disable-icu --disable-iconvv")
+            flags.append("--disable-icu --disable-iconv")
 
-        def add_defines(option, library):
-            if option:
-                for define in self.deps_cpp_info[library].defines:
-                    flags.append("define=%s" % define)
+        def add_defines(library):
+            for define in self.deps_cpp_info[library].defines:
+                flags.append("define=%s" % define)
 
-        if self._zip_bzip2_requires_needed:
-            add_defines(self.options.zlib, "zlib")
-            add_defines(self.options.bzip2, "bzip2")
-            add_defines(self.options.lzma, "xz_utils")
-            add_defines(self.options.zstd, "zstd")
+        if self._with_zlib:
+            add_defines("zlib")
+        if self._with_bzip2:
+            add_defines("bzip2")
+        if self._with_lzma:
+            add_defines("xz_utils")
+        if self._with_zstd:
+            add_defines("zstd")
 
-        if self._is_msvc and self.settings.compiler.runtime:
+        if self._is_msvc:
```

Comment:
in general, I'd leave `self.settings.compiler.runtime` check and omit `_is_msvc` here. why? because other compilers (e.g. clang-cl) may use `self.settings.compiler.runtime`. but this is irrelevant for this PR, so feel free to ignore :)

Reply:
But msvc always has a runtime.
If the test would have been `if self._is_msvc or self.settings.compiler.get_safe("runtime")`, then you would be right.